src/Shell/Task/ModelTask.php

```diff
@@ -395,17 +427,69 @@ public function findBelongsToMany($model, array $associations)
                 $assocTable = substr($otherTableName, 0, $otherOffset);
             }
             if ($assocTable && in_array($assocTable, $tables)) {
-                $habtmName = $this->_camelize($assocTable);
-                $assoc = [
-                    'alias' => $habtmName,
-                    'foreignKey' => $foreignKey,
-                    'targetForeignKey' => $this->_modelKey($habtmName),
-                    'joinTable' => $otherTableName
-                ];
-                if ($assoc && $this->plugin) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+                $assocTableObject = TableRegistry::get($otherTableName);
+
+                $thisForeignKey = $primaryKey;
+                foreach ($thisForeignKey as $i => $fieldName) {
+                    if ($fieldName === 'id') {
+                        $thisForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
+                    }
+                }
+                $otherForeignKeys = $assocTableObject->schema()->columns();
```

Comment:
That's what I thought at first. However looking at your schema AND my migration there is no composite primary key defined. Because of that, none is detected and it is not treated like that. If you set stuff up like `PRIMARY KEY (site_id, user_id, role_id)` then it will work.